### DISCUSSION TITLE
GetTracerProvider returns NoopTracerProvider

### DISCUSSION THREAD

**Discussion by aaronjheng** [open-telemetry/opentelemetry-go#3561] [Q&A] [answered]
Per the comments from `GetTracerProvider`, if no global trace provider is registered, `GetTracerProvider` returns `NoopTracerProvider`. However,  there is always a global trace provider registered. Am I missing something? If so, how can I get a `NoopTracerProvider` provider?
```
// GetTracerProvider returns the registered global trace provider.
// If none is registered then an instance of NoopTracerProvider is returned.
//
// Use the trace provider to create a named tracer. E.g.
//
//	tracer := otel.GetTracerProvider().Tracer("example.com/foo")
//
// or
//
//	tracer := otel.Tracer("example.com/foo")
func GetTracerProvider() trace.TracerProvider {
	return global.TracerProvider()
}
```

**Comment by MrAlias** [comment, marked as answer]
> However, there is always a global trace provider registered.

That is not guaranteed. If you do not explicitly call [`SetTracerProvider`](https://pkg.go.dev/go.opentelemetry.io/otel#SetTracerProvider) no global tracer provider is registered, a noop will be returned in this case.

If you would explicitly like a `NoopTracerProvider` you can create one with [`NewNoopTracerProvider`](https://pkg.go.dev/go.opentelemetry.io/otel/trace#NewNoopTracerProvider).

**Reply by aaronjheng** [reply]
> If you do not explicitly call [SetTracerProvider](https://pkg.go.dev/go.opentelemetry.io/otel#SetTracerProvider) no global tracer provider is registered, a noop will be returned in this case.

You mean `go.opentelemetry.io/otel/internal/global.tracerProvider` is acting like a [NoopTracerProvider](https://pkg.go.dev/go.opentelemetry.io/otel/trace#NoopTracerProvider) but it's not a [NoopTracerProvider](https://pkg.go.dev/go.opentelemetry.io/otel/trace#NoopTracerProvider) in this case. Right?

**Reply by MrAlias** [reply]
Correct, it will delegate all functionality to a noop TracerProvider until a user registers a TracerProvider.

**Reply by aaronjheng** [reply]
Perhaps we could clarify more precisely in comments.
